src/renderers/webgl/WebGLLights.js

```diff
@@ -407,8 +407,21 @@ function WebGLLights() {
 
 		if ( rectAreaLength > 0 ) {
 
-			state.rectAreaLTC1 = UniformsLib.LTC_1;
-			state.rectAreaLTC2 = UniformsLib.LTC_2;
+			if ( capabilities.isWebGL2 || extensions.get( 'OES_texture_float_linear' ) ) {
+
+				state.rectAreaLTC1 = UniformsLib.LTC_FLOAT_1;
+				state.rectAreaLTC2 = UniformsLib.LTC_FLOAT_2;
+
+			} else if ( extensions.get( 'OES_texture_half_float_linear' ) ) {
+
+				state.rectAreaLTC1 = UniformsLib.LTC_HALF_1;
+				state.rectAreaLTC2 = UniformsLib.LTC_HALF_2;
+
+			} else {
+
+				throw 'missing webgl extension';
```

Comment:
Please use `console.error()`.
```js
console.error( 'THREE.WebGLRenderer: Unable to use RectAreaLight. Missing WebGL extensions.' );
```

Reply:
Thanks for all your helpful suggestions, which I've applied. 

I tested this `console.error()` by forcing the previous checks to fail and running it from the `examples/webgl_lights_rectarealight.html` script, but the console error kept printing in an infinite loop. I will investigate this more tomorrow.

Reply:
I think it's okay. I suppose this message will almost never be logged (since the support for `OES_texture_half_float_linear` is very good).